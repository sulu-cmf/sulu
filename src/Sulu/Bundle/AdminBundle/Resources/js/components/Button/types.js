// @flow

export type ButtonSkin = 'primary' | 'secondary' | 'link' | 'icon';
