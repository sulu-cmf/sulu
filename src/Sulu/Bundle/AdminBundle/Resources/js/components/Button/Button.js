// @flow
import React from 'react';
import classNames from 'classnames';
import buttonStyles from './button.scss';

type Props = {
    children: string,
    skin: 'primary' | 'secondary',
    onClick: () => void,
};

export default class Button extends React.PureComponent<Props> {
    handleClick = () => {
        this.props.onClick();
    };

    render() {
        const {
            children,
            skin,
        } = this.props;

        const buttonClass = classNames(buttonStyles.button, buttonStyles[skin]);

        return (
            <button className={buttonClass} onClick={this.handleClick}>
                {children}
            </button>
        );
    }
}
