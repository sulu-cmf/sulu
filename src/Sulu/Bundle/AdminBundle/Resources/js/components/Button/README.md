The `Button` component displays a button.

```javascript
const onClick = () => {
    /* do click things */
    alert('Clicked this nice button, congrats!');
};

<Button
    skin="primary"
    onClick={onClick}>
    Click me dude
</Button>
```

```javascript
const onClick = () => {
    /* do click things */
    alert('Clicked this nice button, congrats!');
};

<Button
    skin="secondary"
    onClick={onClick}>
    Click me dude
</Button>
```

```javascript
const onClick = () => {
    /* do click things */
    alert('Clicked this nice button, congrats!');
};

<Button
    skin="link"
    onClick={onClick}>
    Click me dude
</Button>
```

The buttons can also be used in combination with an icon.

```javascript
<Button skin="primary" icon="su-plus">
    Add something
</Button>
```

```javascript
<Button skin="secondary" icon="su-plus">
    Add something
</Button>
```

```javascript
<Button skin="link" icon="su-plus">
    Add something
</Button>
```

The prop `showDropdownIcon` displays a drop down icon on the right side.

```javascript
<Button skin="icon" icon="su-plus" showDropdownIcon={true} />
```

```javascript
<Button skin="primary" showDropdownIcon={true}>
    Add something
</Button>
```

```javascript
<Button skin="secondary" showDropdownIcon={true}>
    Add something
</Button>
```

```javascript
<Button skin="link" showDropdownIcon={true}>
    Add something
</Button>
```

It's also possible to have a icon and a drop down icon.

```javascript
<Button icon="su-plus" skin="primary" showDropdownIcon={true}>
    Add something
</Button>
```