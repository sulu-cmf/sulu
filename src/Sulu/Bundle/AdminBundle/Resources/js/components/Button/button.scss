@import '../../containers/Application/colors.scss';
@import './variables.scss';

$borderRadius: 3px;
$fontSize: 12px;

$primaryBackground: $shakespeare;
$primaryColor: $white;
$primaryDisabledColor: $silver;

$secondaryBorder: $silver;
$secondaryBackground: $wildSand;
$secondaryColor: $black;
$secondaryDisabledColor: $silver;

$linkFontSize: 14px;
$linkColor: $black;

$iconFontSize: 18px;
$iconBackgroundColorActive: $doveGray;
$iconBackgroundColorHover: $mercury;
$iconColorActive: $white;

.button {
    position: relative;
    white-space: nowrap;

    &:disabled {
        cursor: default;
    }
}

.primary,
.secondary {
    font-size: $fontSize;
    border: 1px solid;
    border-radius: $borderRadius;
    line-height: 28px;
    cursor: pointer;
    text-align: center;
    min-width: 140px;
    padding: 0 40px;

    .button-icon {
        display: inline-flex;
        align-items: center;
        position: absolute;
        top: 0;
        bottom: 0;
        left: 10px;
    }
}

.primary {
    border-color: $primaryBackground;
    background: $primaryBackground;
    color: $primaryColor;

    &:disabled {
        border-color: $primaryDisabledColor;
        background-color: $primaryDisabledColor;
    }
}

.secondary {
    border-color: $secondaryBorder;
    background: $secondaryBackground;
    color: $secondaryColor;

    &:disabled {
        color: $secondaryDisabledColor;
    }
}

.link {
    padding: 0;
    margin: 0 10px 0 0;
    border: 0;
    background: transparent;
    cursor: pointer;
    font-size: $linkFontSize;
    color: $linkColor;

    .text {
        text-decoration: underline;
    }

    .button-icon {
        margin-right: 10px;
    }
}

.icon {
    background: none;
    width: 30px;
    height: 30px;
    border-radius: $iconBorderRadius;
    border: 1px solid $iconBorderColor;
    color: $iconBorderColor;
    cursor: pointer;
    font-size: $iconFontSize;

    &:hover {
        background-color: $silver;
    }

    &.active {
        background-color: $iconBorderColor;
        color: $iconColorActive;
    }
}

.icon-span {
    line-height: 28px;
}

.loader {
    display: none;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}

.loading {
    background-color: transparent;
    border-color: transparent;

    .text {
        opacity: 0;
    }

    .loader {
        display: flex;
    }
}
