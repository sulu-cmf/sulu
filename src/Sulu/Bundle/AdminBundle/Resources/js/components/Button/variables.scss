@import '../../containers/Application/colors.scss';

$iconBorderColor: $doveGray;
$iconBorderRadius: 4px;
