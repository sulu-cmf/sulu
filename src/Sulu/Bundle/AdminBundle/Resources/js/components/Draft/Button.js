// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import type {Button as ButtonProps} from './types';
import buttonStyles from './button.scss';

type Props = ButtonProps & {
    onToggle: (handle: string, style: string) => void,
    active?: boolean,
};

export default class Button extends React.Component<Props> {
    handleOnMouseDown = (event: SyntheticEvent<HTMLButtonElement>) => {
        event.preventDefault();
        this.props.onToggle(this.props.handle, this.props.style);
    };

    handleClick = (event: SyntheticEvent<HTMLButtonElement>) => {
        event.preventDefault();
    };

    render() {
        const {
            icon,
            active,
        } = this.props;

        const buttonClass = classNames(
            buttonStyles.button,
            {
                [buttonStyles.active]: active,
            }
        );

        return (
            <button
                className={buttonClass}
                onMouseDown={this.handleOnMouseDown}
                onClick={this.handleClick}
            >
                <Icon name={icon} />
            </button>
        );
    }
}
