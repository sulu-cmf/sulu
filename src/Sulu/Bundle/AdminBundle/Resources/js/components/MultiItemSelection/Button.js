// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import buttonStyles from './button.scss';

type Props = {
    icon: string,
    onClick: () => void,
    location: 'left' | 'right',
};

export default class Button extends React.PureComponent<Props> {
    handleClick = () => {
        this.props.onClick();
    };

    render() {
        const {
            icon,
            location,
        } = this.props;
        const buttonClass = classNames(
            buttonStyles.button,
            buttonStyles[location]
        );

        return (
            <button
                type="button"
                onClick={this.handleClick}
                className={buttonClass}
            >
                <Icon name={icon} />
            </button>
        );
    }
}
