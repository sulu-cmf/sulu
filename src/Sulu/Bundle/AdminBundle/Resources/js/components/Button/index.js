// @flow
import Button from './Button';
import type {ButtonSkin} from './types';

export default Button;
export type {ButtonSkin};
