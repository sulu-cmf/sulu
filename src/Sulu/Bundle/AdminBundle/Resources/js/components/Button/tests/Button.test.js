/* eslint-disable flowtype/require-valid-file-annotation */
import React from 'react';
import {render, shallow} from 'enzyme';
import Button from '../Button';

test('Button should render with skin primary', () => {
    expect(render(<Button skin="primary" />)).toMatchSnapshot();
});

test('Button should render with skin secondary', () => {
    expect(render(<Button skin="secondary" />)).toMatchSnapshot();
});

test('Button should render with skin link', () => {
    expect(render(<Button skin="link" />)).toMatchSnapshot();
});

test('Button should call the callback on click', () => {
    const onClick = jest.fn();
    const button = shallow(<Button skin="primary" onClick={onClick} />);
    button.find('button').simulate('click');
    expect(onClick).toBeCalled();
});
