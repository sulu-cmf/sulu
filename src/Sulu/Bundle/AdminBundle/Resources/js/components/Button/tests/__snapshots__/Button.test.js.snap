// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the button with icon 1`] = `
<button
  class="button secondary"
  type="button"
>
  <span
    aria-label="su-plus"
    class="su-plus buttonIcon"
  />
  <span
    class="text"
  >
    Add something
  </span>
</button>
`;

exports[`Should render with skin icon 1`] = `
<button
  class="button icon"
  type="button"
/>
`;

exports[`Should render with skin icon and active 1`] = `
<button
  class="button icon active"
  type="button"
/>
`;

exports[`Should render with skin link 1`] = `
<button
  class="button link"
  type="button"
/>
`;

exports[`Should render with skin primary 1`] = `
<button
  class="button primary"
  type="button"
/>
`;

exports[`Should render with skin secondary 1`] = `
<button
  class="button secondary"
  type="button"
/>
`;

exports[`should render disabled with skin secondary 1`] = `
<button
  class="button secondary"
  disabled=""
  type="button"
/>
`;
